File: typescript/ARS_SHELL_CRYPT/security/SecurityLevel.ts
import ARS_SHELL_CRYPT from "../cryptage/ARS_SHELL_CRYPT";
import { CHAR_LIST } from "../standard/StandardList";
import { mt_rand } from "../utils/Utils";

interface ISecData {
    spacing: number;
    firstStrBreak: string;
    secondStrBreak: string;
}

export default class SecurityLevel extends ARS_SHELL_CRYPT {

    public sec_data: ISecData;

    /**
     * Generate a string follow the Encryption level
     * @param sec_lvl
     */
    public gen_sec_str(sec_lvl: number): string {
        let sec_str = "`";

        for (let i = 0; i < sec_lvl * 10; i++) {
            sec_str += CHAR_LIST[mt_rand(0, this.count() - 1)];
        }
        return sec_str;
    }

    /**
     * Serialize the data
     * @param sec_lvl
     */
    public serialize_sec_data(sec_lvl: number) {
        return this.sec_data = {
            spacing: sec_lvl * 5,
            firstStrBreak: this.gen_sec_str(sec_lvl),
            secondStrBreak: this.gen_sec_str(sec_lvl),
        }
    }
}

File: python/ARS_SHELL_CRYPT/security/SecurityLevel.py
import random
from cryptage.ARS_SHELL_CRYPT import ARS_SHELL_CRYPT


class SecurityLevel(ARS_SHELL_CRYPT):

    def gen_sec_string(self, sec_lvl):
        sec_str = "`"

        for i in range(sec_lvl * 10):
            sec_str += super().CHAR_LIST[random.randint(0, len(super().CHAR_LIST) - 1)]
        return sec_str

    def serialize_sec_data(self, sec_lvl):
        if sec_lvl <= 4:
            return {
                "spacing": sec_lvl * 10,
                "first_str_break": self.gen_sec_string(sec_lvl),
                "second_str_break": self.gen_sec_string(sec_lvl)
            }
        else:
            return {
                "spacing": sec_lvl * 9,
                "first_str_break": self.gen_sec_string(sec_lvl),
                "second_str_break": self.gen_sec_string(sec_lvl)
            }